src/db.ts

```diff
@@ -1,13 +1,54 @@
 // TODO: Extract to library
-import { customType } from 'drizzle-orm/pg-core';
+import { customType, pgTable, pgSchema } from 'drizzle-orm/pg-core';
 import { drizzle as drizzleNeon } from 'drizzle-orm/neon-http';
 import { drizzle as drizzlePostgres } from 'drizzle-orm/node-postgres';
 import { type DrizzleConfig } from 'drizzle-orm';
 import { Address, Uint, Int, Bytes } from './types';
 import { Context } from 'hono';
 import { Pool, escapeIdentifier } from 'pg';
 
+// Get the name of the schema associated with this deployment
+export function schemaName(c: Context): string | null {
+	const searchPath = parseSearchPath(c.env.DB_CONNECTION_STRING);
+	if (!searchPath || searchPath.length === 0) {
+		return null;
+	}
+	return searchPath[0] ?? null;
+}
+
+// Qualify a table name for this deployment
+export function tableName(c: Context, name: string): string {
+	const schema = schemaName(c);
+	if (!schema) {
+		return name;
+	}
+	return `${schema}.${escapeIdentifier(name)}`;
+}
+
+type Tail<T extends any[]> = T extends [any, ...infer R] ? R : never;
```

Comment:
Any reason you didn't use https://orm.drizzle.team/docs/schemas ? 

I think that would handle the escaping etc. for you.